docs/developer/policy-engine.md

```diff
@@ -0,0 +1,248 @@
+# Policy Engine
```

Comment:
Probably gets out of the scope here since it concerns "how to set up an extension" (that's missing from the doc currently), but since there are the details on how to set up the build file, probably we need to specify that a `ServiceExtension` file in the `META-INF/services` directory is needed

Reply:
@ndr-brt I applied you feedback.